### ISSUE TITLE
"extensible web" approach to nav-rule

### ISSUE THREAD

**Issue by frivoal** [WICG/spatial-navigation#6]
The generic spat-nav heuristic can fail, and manually coding everything with `nav-up` / `nav-down` / `nav-right` / `nav-left` can be tedious, so something along the lines of the proposed `nav-rule` seems very useful.

However, I think coming up with the correct design for this property will be very difficult. Instead
I think we should provide some JS primitives to let page authors and framework authors come up with their own heuristics or algorithms, and only once some clear patterns emerge, standardize that.

I think that all we'd need would be a way for the author to register a JS function that would be called back when the user attempts to navigate directionally, letting javascript figure out which element the focus should go to, and call `focus()` on it (and `preventDefault()`).

To avoid the tedium of having to set this individually on each and every element the user may need to navigate away from, this should be possible to register on any element in order to get a callback when the user tries to navigate away from any focusable element within that subtree.

**Comment by frivoal** [comment]
(BTW, by "extensible web", I am referring to this: https://github.com/extensibleweb/manifesto)

**Comment by frivoal** [comment]
Closing in favor of #2
